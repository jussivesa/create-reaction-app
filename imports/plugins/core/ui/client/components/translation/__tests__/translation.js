import React from "react";
import { shallow } from "enzyme";
import shallowToJSON from "enzyme-to-json";
import Translation from "../translation";

jest.mock("/client/api", () => ({
  i18next: {
    t: (key, { defaultValue }) => defaultValue || key
  }
}));


/**
 * Order Summary is a display only component
 * It receives props and displays them accordingly
 */

afterEach(() => {
  jest.clearAllMocks();
});

/**
 * Snapshots make sure your UI does not change unexpectedly
 */

test("Translation snapshot test", () => {
  const component = shallow((
    <Translation
      defaultValue="Translated Text"
      i18nKey={"path.to.key"}
    />
  ));
  const tree = shallowToJSON(component);
  expect(tree).toMatchSnapshot();
});
