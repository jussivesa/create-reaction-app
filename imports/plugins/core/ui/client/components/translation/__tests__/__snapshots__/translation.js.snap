// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Translation snapshot test 1`] = `
<span>
  Translated Text
</span>
`;
